### PR TITLE
Update cost model

### PR DESCRIPTION
**Pull request by pontusmelke**
The relative costs of operators have changed quite some bit since 2.3 and the
cost model should be updated to accommodate this.

### CODE REVIEW

File: community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/logical/CardinalityCostModel.scala [hunk 1]
```diff
@@ -38,16 +38,20 @@ object CardinalityCostModel extends CostModel {
...
 
     case _: Expand |
-         _: VarExpand  => 2.5
+         _: VarExpand  => 2.0
+
+    case _: NodeUniqueIndexSeek |
+         _: NodeIndexSeek |
+         _: NodeIndexContainsScan |
+         _: NodeIndexEndsWithScan |
+         _: NodeIndexScan => 3.0
```
Comment (Mats-SX) [inline community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/logical/CardinalityCostModel.scala:54]: Given that this file has so many magic constants now, does it even make sense to have some of them named? Ie, `FAST_STORE`, `CPU_BOUND` etc. They seem as arbitrary/measured as these new ones.

**Comment by Mats-SX** [comment]
@pontusmelke This looks nice. I guess we will see what the benchmark says after this is merged to see if our simulation program works.

File: community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/logical/CardinalityCostModel.scala (hunk 1 above)
Comment (pontusmelke) [inline community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/logical/CardinalityCostModel.scala:54]: No I agree, we should probably just get rid of them all.

**Comment by pontusmelke** [comment]
@Mats-SX Yes if the benchmarks doesn't like this we'll just revert. I have run some of them locally and haven't seen any regressions.

**Pull request merged**